cmd/etcd-launcher/main.go

```diff
@@ -130,6 +214,139 @@ func etcdCmd(config *envConfig) []string {
+	return err
+}
+
+func (e *etcdCluster) getClientWithEndpoints(eps []string) (*clientv3.Client, error) {
+	var err error
+	for i := 0; i < 5; i++ {
+		cli, err := clientv3.New(clientv3.Config{
+			Endpoints:   eps,
+			DialTimeout: 2 * time.Second,
+		})
+		if err == nil && cli != nil {
+			return cli, nil
+		}
+		time.Sleep(5 * time.Second)
+	}
+	return nil, fmt.Errorf("failed to establish client connection: %v", err)
+
+}
+
+func (e *etcdCluster) listMembers() ([]*etcdserverpb.Member, error) {
+	if e.client == nil {
+		return nil, fmt.Errorf("can't find cluster client")
+	}
+	resp, err := e.client.MemberList(context.Background())
+	if err != nil {
+		return nil, err
+	}
+	return resp.Members, err
+}
+
+func (e *etcdCluster) isClusterMember(name string) (bool, error) {
+	members, err := e.listMembers()
+	if err != nil {
+		return false, err
+	}
+	if len(members) == 0 {
+		return false, nil
+	}
+
+	for _, member := range members {
+		if member.Name == name || member.PeerURLs[0] == fmt.Sprintf("http://%s.etcd.%s.svc.cluster.local:2380", e.config.podName, e.config.namespace) {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (e *etcdCluster) isHealthy() (bool, error) {
+	if err := e.getClient(); err != nil {
+		return false, err
+	}
+	// just get a key from etcd, this is how `etcdctl endpoint health` works!
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	_, err := e.client.Get(ctx, "healthy")
+	cancel()
+	if err != nil && err != rpctypes.ErrPermissionDenied {
+		return false, nil
+	}
+	return true, nil
+}
+
+func (e *etcdCluster) isEndpointHealthy(endpoint string) (bool, error) {
+	client, err := e.getClientWithEndpoints([]string{endpoint})
+	if err != nil {
+		return false, err
+	}
+	// just get a key from etcd, this is how `etcdctl endpoint health` works!
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	_, err = client.Get(ctx, "healthy")
+	cancel()
+	if err != nil && err != rpctypes.ErrPermissionDenied {
+		return false, nil
+	}
+	return true, nil
+
+}
+
+func (e *etcdCluster) isLeader() (bool, error) {
+	var err error
+	if err = e.getLocalClient(); err != nil {
+		return false, err
+	}
+	var resp *clientv3.StatusResponse
+	for i := 0; i < 10; i++ {
+		resp, err = e.localClient.Status(context.Background(), e.endpoint())
+		if err != nil || resp.Leader == 0 {
+			time.Sleep(2 * time.Second)
+			continue
+		}
+		if resp.Header.MemberId == resp.Leader {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (e *etcdCluster) removeDeadMembers() error {
+	members, err := e.listMembers()
+	if err != nil {
+		return err
+	}
+	for _, member := range members {
+		if member.Name == e.config.podName {
+			continue
+		}
+		healthy, err := e.isEndpointHealthy(member.PeerURLs[0])
+		if err != nil {
+			continue
+		}
+		if !healthy {
+			if _, err := e.client.MemberRemove(context.Background(), member.ID); err != nil {
```

Comment:
you don't need the check here and you are not wrap the error, thus remove the if statement please and return the method. 
```
return e.client.MemberRemove(context.Background(), member.ID)
```